app/cli/cmd/root.go

```diff
@@ -83,6 +93,38 @@ func NewRootCmd(l zerolog.Logger) *cobra.Command {
 			return nil
 		},
 		PersistentPostRunE: func(cmd *cobra.Command, args []string) error {
+			// Do not track telemetry if the env variable is set
+			var telemetryClient telemetry.Client
+			if os.Getenv(doNotTrackEnv) == "1" || os.Getenv(doNotTrackEnv) == "true" {
+				logger.Debug().Msg("Telemetry is disabled")
+			} else {
+				var err error
+				telemetryClient, err = posthog.NewClient(posthogAPIKey, posthogEndpoint)
+				if err != nil {
+					logger.Debug().Err(err).Msgf("creating telemetry client: %v", err)
+				}
+			}
+
+			cmdTracker := telemetry.NewCommandTracker(telemetryClient)
+
+			var cmdHierarchy []string
+			currentCmd := cmd
+			// While the current command is not the root command, keep iteration.
+			// This is done to get the full hierarchy of the command and remove the root command from the hierarchy.
+			for currentCmd.Use != "chainloop" {
+				cmdHierarchy = append([]string{currentCmd.Use}, cmdHierarchy...)
+				currentCmd = currentCmd.Parent()
+			}
+
+			cmdLine := strings.Join(cmdHierarchy, " ")
+
+			err := cmdTracker.Track(cmd.Context(), cmdLine, telemetry.Tags{
+				"cli_version": Version,
```

Comment:
it would be great if we could also capture the conttrolplaneURL but hash it anonymously. We have the controlplane URI in viper, I guess we should be able to find it.

Reply:
and also the command status. Isn't it a way to detect if the command returned an error? We capture errors somewhere else globally to do some manual handling, i.e show you that you need to re-login

Reply:
I have added both of them computing the hash of the control plane url